napalm_ansible/modules/napalm_install_config.py

```diff
@@ -291,6 +293,13 @@ def main():
     except Exception as e:
         module.fail_json(msg="cannot diff config: " + str(e))
 
+    try:
```

Comment:
this is the same pattern as with `archive_file`, which is fine and means we probably should have a small function. Something like:

```
def _save_file(which, where):
   if which is not None:
        config = device.get_config(retrieve=which)[which]
        save_to_file(config_config, where)
```

And then just do:

```
try:
    _save...
except Exception as e:
     module.fail_json(...)
```

Reply:
It's a sound plan but seems a little over eager to modularise IMHO. I find this much more readable at first glance. I'll do as you say though if you like :)